### PR TITLE
fix: make assume role optional

### PR DESCRIPTION
**Pull request by slesaad**
**Summary:** Summary of changes

Assume role is supposed to be optional.

## Changes

Make assume role optional

### CODE REVIEW

**Comment by smohiudd** [comment]
@slesaad I believe the `BACKEND_VPC_ID` was added because in UAH we have two VPCs - one for airflow and one for everything else. I agree we should probably have one VPC and will need to redeploy dev to use the `veda-shared-base/network/vpc` VPC and exclude the airflow VPC

File: infrastructure/main.tf [hunk 1]
```diff
@@ -224,10 +181,7 @@ resource "aws_security_group" "workflows_api_handler_sg" {
   lifecycle { create_before_destroy = true }
 }
 
-# Function to build the JWKS URL
-locals {
-  build_jwks_url = "${format("https://cognito-idp.%s.amazonaws.com/%s/.well-known/jwks.json", local.aws_region, var.userpool_id)}"
-}
```
Comment (smohiudd) [inline infrastructure/main.tf:230]: Is this not used anywhere? that's why it was removed?

File: infrastructure/locals.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    conditional_workflows_lambda_policy = var.data_access_role_arn != "" ? concat(local.workflows_lambda_policy, [
+        {
+        Effect: "Allow",
+        Action: ["sts:AssumeRole"],
+        Resource: [var.data_access_role_arn]
+        }
+    ]) : local.workflows_lambda_policy
+
+  build_jwks_url = "${format("https://cognito-idp.%s.amazonaws.com/%s/.well-known/jwks.json", local.aws_region, var.userpool_id)}"
```
Comment (slesaad) [inline infrastructure/locals.tf:52]: `build_jwks_url` was moved here @smohiudd ☝️

File: infrastructure/main.tf (hunk 1 above)
Comment (slesaad) [reply infrastructure/main.tf:230]: https://github.com/NASA-IMPACT/veda-data-airflow/pull/156#discussion_r1613733005

**Comment by slesaad** [comment]
> @slesaad I believe the `BACKEND_VPC_ID` was added because in UAH we have two VPCs - one for airflow and one for everything else. I agree we should probably have one VPC and will need to redeploy dev to use the `veda-shared-base/network/vpc` VPC and exclude the airflow VPC

hmm - then let me update this to use `VPC_ID` as the default for `BACKEND_VPC_ID`

File: infrastructure/locals.tf (hunk 2 above)
Comment (smohiudd) [reply infrastructure/locals.tf:52]: oh missed that. Thanks!

**Review by smohiudd — APPROVED** [review]

**Review by ividito — APPROVED** [review]

**Pull request merged**